cirq-web/cirq_ts/e2e/bloch_sphere/bloch_sphere_e2e_test.ts

```diff
@@ -0,0 +1,140 @@
+</html>
+`;
+
+const newVectorBrowserContent = `
+<!doctype html>
+<meta charset="UTF-8">
+<html lang="en">
+    <head>
+    <title>Cirq Web Development page</title>
+    </head>
+    <body>
+    <div id="container"></div>
+    <script>${bundle_string}</script>
+    <script>
+        renderBlochSphere('{"radius": 5}', 'container').addVector('{"x": 1,"y": 1, "z": 2, "length": 5}');
+    </script>
+    </body>
+</html>
+`;
+
+/**
+ * Testing to see if they look the same.
+ */
+
+// Automatically track and cleanup files on exit
+temp.track();
+
+describe('Check Bloch Sphere looks correct', () => {
+  // Create the temporary directory first, then run everything.
+  temp.mkdir('tmp', (err, dirPath) => {
+    const output_path = path.join(dirPath, 'bloch_sphere.png');
+    const new_vector_output_path = path.join(dirPath, 'bloch_sphere_vec.png');
+
+    before(async () => {
+      //Opens a headless browser with the generated HTML file and takes a screenshot.
+      const browser = await puppeteer.launch();
+      const page = await browser.newPage();
+
+      // Take a screenshot of the first image
+      await page.setContent(browserContent);
+      await page.screenshot({path: output_path});
+      await browser.close();
+    });
+
+    it('Bloch sphere in the default state (|0⟩) matches the standard PNG copy', () => {
+      const expected = PNG.PNG.sync.read(
+        readFileSync('e2e/bloch_sphere/bloch_sphere_expected.png')
+      );
+      const actual = PNG.PNG.sync.read(readFileSync(output_path));
+      const {width, height} = expected;
+      const diff = new PNG.PNG({width, height});
+
+      const pixels = pixelmatch(
+        expected.data,
+        actual.data,
+        diff.data,
+        width,
+        height,
+        {threshold: 0.1}
+      );
+      // Accouting for the difference in unicode symbols in browsers
+      expect(pixels).to.be.below(700);
+    });
+
+    before(async () => {
+      //Opens a headless browser with the generated HTML file and takes a screenshot.
+      const browser = await puppeteer.launch();
+      const page = await browser.newPage();
+
+      // Take a screenshot of the second image
+      await page.setContent(newVectorBrowserContent);
+      await page.screenshot({path: new_vector_output_path});
+      await browser.close();
+    });
+
+    it('Bloch sphere with custom statevector matches the standard PNG copy', () => {
+      const expected = PNG.PNG.sync.read(
+        readFileSync('e2e/bloch_sphere/bloch_sphere_expected_custom.png')
+      );
+      const actual = PNG.PNG.sync.read(readFileSync(new_vector_output_path));
+      const {width, height} = expected;
+      const diff = new PNG.PNG({width, height});
+
+      const pixels = pixelmatch(
+        expected.data,
+        actual.data,
+        diff.data,
+        width,
+        height,
+        {threshold: 0.1}
+      );
+      // Accouting for the difference in unicode symbols in browsers
+      expect(pixels).to.be.below(700);
```

Comment:
I'm not sure I get this - I think this should be 0, can you explain a bit more what's happening?

Reply:
This was something I explored and then forgot about...I just fixed it. Good lesson don't change the tests.

So Chromium is weird..what I think happened is that when the browser instance starts, a alert comes from the top of the screen that says "Chromium is being ran under testing" or something and then when puppeteer takes a screenshot, it's getting that image along with the bloch sphere. I was failing my tests a while ago due to it and looked at example screenshots and thought the symbols looked different in different browsers so documented it and then moved on. I adding an argument to the `puppeteer.launch()` function, and now it works with equal 0.